AEPCore/Sources/core/MobileCoreAPI.swift

```diff
@@ -0,0 +1,413 @@
+//
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+*/
+
+import Foundation
+import AEPServices
+
+/// Core APIs for interacting with specific instances of the Adobe Experience Platform SDK.
+@objc(AEPMobileCoreAPI)
+public class MobileCoreAPI: NSObject {
+
+    let instanceIdentifier: String
+    let sdkInstanceIdentifier: SDKInstanceIdentifier?
+    let eventHubProvider: EventHubProvider
+    let logTag: String
+
+    internal init(instanceIdentifier: String, eventHubProvider: EventHubProvider) {
+        self.instanceIdentifier = instanceIdentifier
+        sdkInstanceIdentifier = SDKInstanceIdentifier(id: instanceIdentifier)
+        self.eventHubProvider = eventHubProvider
+        logTag = "MobileCoreAPI-\(sdkInstanceIdentifier?.id ?? instanceIdentifier)"
+        
+    }
+    
+    private func getEventHub(_ debugInfo: String) -> EventHub? {
+        guard let sdkInstanceIdentifier = sdkInstanceIdentifier else {
+            Log.warning(label: logTag, "MobileCoreAPI::\(debugInfo) - SDK instance name \(instanceIdentifier) is invalid.")
+            return nil
+        }
+        
+        guard let eventHub = eventHubProvider.getEventHub(for: sdkInstanceIdentifier) else {
+            Log.warning(label: logTag, "MobileCoreAPI::\(debugInfo) - SDK instance \(instanceIdentifier) has not been initialized.")
```

Comment:
nit: reword to "could not be initialized" as "has not been" seems to imply that it still could be initialized in the future.

Reply:
The initialization may occur later. The error is thrown at this point so the SDK can return a non-nil instance for `MobileCore.api(for: instance)`, simplifying SDK usage.